pkg/istio-agent/sds-agent.go

```diff
@@ -503,3 +517,9 @@ func applyEnvVars() {
 	}
 	workloadSdsCacheOptions.OutputKeyCertToDir = serverOptions.OutputKeyCertToDir
 }
+
+// shouldProvisionCertificates returns true if certs needs to be provisioned for the workload/gateway.
+// Returns flase, when proxy uses file mounted certificates i.e. CA_PROVIDER is empty.
```

Comment:
Same as my previous comments, caProviderEnv should not be reused to express whether proxy uses file mounted certs or not. I suggest two alternatives: 
1. Detecting whether proxy uses file mounted certs or not.
2. Use a new variable to represent whether proxy uses file mounted certs or not.